.github/workflows/ghalint.yml

```diff
@@ -0,0 +1,69 @@
+name: ghalint
+
+on:
+  pull_request:
+  merge_group:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  setup-job:
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    permissions:
+      contents: read
+      pull-requests: read
+    outputs:
+      has-changes: ${{ steps.changes.outputs.has-changes }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+      - uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changes
+        with:
+          filters: |
+            has-changes:
+              - '.github/workflows/ghalint.yml'
+              - '.github/workflows/**/*.yml'
+              - '.github/workflows/**/*.yaml'
+
+  ghalint:
+    needs: [setup-job]
+    if: ${{ needs.setup-job.outputs.has-changes == 'true' }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+      
+      - name: Download ghalint
+        run: |
+          curl -L -o ghalint.tar.gz https://github.com/suzuki-shunsuke/ghalint/releases/download/v1.5.1/ghalint_1.5.1_linux_amd64.tar.gz
+          tar -xzf ghalint.tar.gz
+          chmod +x ghalint
+          sudo mv ghalint /usr/local/bin/
+          
+      - name: Run ghalint
+        run: |
+          # Run ghalint and filter for only action_ref_should_be_full_length_commit_sha policy
+          set +e  # Don't exit on error immediately
+          
+          # Run ghalint and capture output
+          ghalint run 2>&1 | tee ghalint_output.txt
+          
+          # Check for action_ref_should_be_full_length_commit_sha errors
+          echo "=== Checking for action_ref_should_be_full_length_commit_sha violations ==="
+          if grep "action_ref_should_be_full_length_commit_sha" ghalint_output.txt; then
+            echo "❌ Found action_ref_should_be_full_length_commit_sha violations!"
+            exit 1
+          else
+            echo "✅ No action_ref_should_be_full_length_commit_sha violations found"
+            exit 0
+          fi
```

Comment:
It appears that ghalint's ability to ignore lint is incomplete and not all rules can be excluded.Also, the rules that could be excluded were not necessary because they were not covered in our implementation.

ref: https://github.com/suzuki-shunsuke/ghalint/tree/main?tab=readme-ov-file#disable-policies

Therefore, we implemented the exclusion process on the job side as a workaround.
The results can be seen here: 

- Cases where action_ref_should_be_full_length_commit_sha fails: https://github.com/liam-hq/liam/actions/runs/15870057039/job/44744378667?pr=2198 
- Other cases where lint fails: https://github.com/liam-hq/liam/actions/runs/15870111155/job/44744534855?pr=2198